### PR TITLE
Add new sleep sensors for new sleep API

### PR DESCRIPTION
**Pull request by dshokouhi**
## Summary

Fixes: #1383 

Uses the new Sleep API to make 2 new sensors.  1 for sleep confidence and the other for sleep segment. These sensors are for the full version only as it relies on Google services.

https://developers.google.com/location-context/sleep

## Screenshots

image

image

image

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#467

### CODE REVIEW

File: app/src/full/java/io/homeassistant/companion/android/sensors/ActivitySensorManager.kt [hunk 1]
```diff
@@ -119,16 +190,52 @@ class ActivitySensorManager : BroadcastReceiver(), SensorManager {
...
+
+            Log.d(TAG, "Registering for activity updates.")
+            actReg.requestActivityUpdates(120000, pendingIntent)
+        }
+        if (isEnabled(context, sleepConfidence.id) || isEnabled(context, sleepSegment.id)) {
+            val pendingIntent = getSleepPendingIntent(context)
+            Log.d(TAG, "Unregistering for sleep updates")
+            ActivityRecognition.getClient(context).removeSleepSegmentUpdates(pendingIntent)
+            Log.d(TAG, "Registering for sleep updates")
+            val task = when {
```
Comment (JBassett) [inline app/src/full/java/io/homeassistant/companion/android/sensors/ActivitySensorManager.kt:207]: Do we really want to unregister then reregister every time we get a requested update?

File: app/src/full/java/io/homeassistant/companion/android/sensors/ActivitySensorManager.kt [hunk 2]
```diff
@@ -119,16 +190,32 @@ class ActivitySensorManager : BroadcastReceiver(), SensorManager {
...
+                SleepSegmentRequest.getDefaultSleepSegmentRequest()
+            )
+            task.addOnSuccessListener {
+                Log.d(TAG, "Successfully registered for sleep updates")
+            }
+            task.addOnFailureListener {
+                Log.e(TAG, "Failed to register for sleep updates", it)
+                ActivityRecognition.getClient(context).removeSleepSegmentUpdates(pendingIntent)
+            }
+        }
```
[the remark this answers is not in the document]
Comment (dshokouhi) [reply app/src/full/java/io/homeassistant/companion/android/sensors/ActivitySensorManager.kt]: I suppose we can just keep track of registration and only register if we need to.  I was honestly following along the other sensor here.  I will add a variable to keep track of registration here

**Comment by dshokouhi** [comment]
I noticed that the classify events were delayed and they come pretty frequently so sending those over as soon as we get them. The sleep segment events show up once or twice a day so didn't see a need to send those over immediately especially considering the state may not change frequently and we can get a lot of events with the same data.  Better to keep this one separate.

**Comment by dshokouhi** [comment]
Some overnight graphs to get an idea of what the data looks like.  Sleep confidence shows up about every 5-15 min or so.

image

Sleep segment shows up about once a day or so
image

Not sure why the status shows as missing data but thats the status that was given.

**Review by JBassett — APPROVED** [review]

**Pull request merged**